### ISSUE TITLE
Private properties support

### ISSUE THREAD

**Issue by vitaly-t** [TypeStrong/typedoc#1229]
TypeScript supports private properties since v3.8.0, which this library fails to recognize:

```ts
class Test {
    /**
         private property example.
    */
    #prop: number = 123;
}
```

This library throws ` Invalid character` error, which stops us from using this new TypeScript feature, because then we cannot generate documentation.

I'm not sure if this shouldn't be labelled as a bug. At least, this should be a high-priority feature/fix.

**Comment by Gerrit0** [comment]
Workarounds:
1. Use `typedoc@next` - TS is moving to a peer dependency, but the library mode feature isn't done yet so not a full release.
2. Remove `node_modules/typedoc/node_modules/typescript` to force TypeDoc to use your version of TS. Properties will show up documented with the name `#name` (but I don't think they are correctly marked private)

I should have time this weekend to pull the parts of my original plan for 0.17 that are done into master and re-schedule library mode for 0.18.

**Comment by vitaly-t** [comment]
Thank you for quick response!

For now, as a work-around, we use `WeakMap` instead, which is what TypeScript transpiles private properties into (when generating es5 code).

**Comment by Gerrit0** [comment]
Fixed with 5d41a2e03a4cefa29ae84e7f9ef409d976d4ac56, release as soon as I get done going through issues and making sure I got everything.
